disco_aws_automation/disco_metanetwork.py

```diff
@@ -118,18 +115,40 @@ def _create_security_group(self):
         return security_group
 
     @property
-    def subnets(self):
-        '''Finds or creates the subnets for our metanetwork'''
-        if not self._subnets:
-            self._subnets = find_or_create(
-                self._find_subnets, self._create_subnets
-            )
-        return self._subnets
+    def disco_subnets(self):
+        '''Creates the subnets for our metanetwork'''
+        if not self._disco_subnets:
+            self._disco_subnets = self._create_subnets()
+        return self._disco_subnets
 
-    def _find_subnets(self):
-        return self.vpc.vpc.connection.get_all_subnets(
-            filters=self._resource_filter
-        )
+    def add_nat_gateways(self, allocation_ids):
+        """
+        Creates a NAT gateway in each of the metanetwork's subnet
+        :param allocation_ids: Allocation ids of the Elastic IPs that will be
+                               associated with the NAT gateways
+        """
+        if len(self.disco_subnets.values()) != len(allocation_ids):
+            raise EIPConfigError("The number of subnets does not match with the "
+                                 "number of NAT gateway EIPs provided for {0}: "
+                                 "{1} != {2}"
+                                 .format(self._resource_name(),
+                                         len(self.disco_subnets.values()),
+                                         len(allocation_ids)))
+
+        if self.centralized_route_table:
```

Comment:
This is to handle the case where we are trying to update existing live environments for backward compatibility's sake. If centralized route table is used, it would make a copy of it for each subnet and then delete it.